src/main/app/access-code/AccessCodePostController.ts

```diff
@@ -0,0 +1,54 @@
+import autobind from 'autobind-decorator';
+import { Response } from 'express';
+
+import { getNextStepUrl } from '../../steps';
+import { getCaseApi } from '../case/CaseApi';
+import { CITIZEN_LINK_APPLICANT_2 } from '../case/definition';
+import { AppRequest } from '../controller/AppRequest';
+import { AnyObject, PostController } from '../controller/PostController';
+import { Form } from '../form/Form';
+
+@autobind
+export class AccessCodePostController extends PostController<AnyObject> {
+  constructor(protected readonly form: Form) {
+    super(form);
+  }
+
+  public async post(req: AppRequest<AnyObject>, res: Response): Promise<void> {
+    req.locals.api = getCaseApi(req.session.user, req.locals.logger);
```

Comment:
don't you need to do this as a caseworker? The user won't have access yet

Reply:
I make the call in the case-api with the caseworker, but I need this user token so I know who to link the case role to

Reply:
At this point the respondent would have created their idam user profile and been redirected to the enter-your-access-code page as we would have that as the redirect link as part of the link in the email they receive.